akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -0,0 +1,399 @@
+   * by invoking [[akka.actor.Cancellable]] `cancel` on the returned
+   * handle.
+   */
+  def schedule[U](delay: FiniteDuration, target: ActorRef[U], msg: U): akka.actor.Cancellable
+
+  /**
+   * This Actor’s execution context. It can be used to run asynchronous tasks
+   * like [[scala.concurrent.Future]] combinators.
+   */
+  implicit def executionContext: ExecutionContextExecutor
+
+  /**
+   * Create a child actor that will wrap messages such that other Actor’s
+   * protocols can be ingested by this Actor. You are strongly advised to cache
+   * these ActorRefs or to stop them when no longer needed.
+   *
+   * The name of the child actor will be composed of a unique identifier
+   * starting with a dollar sign to which the given `name` argument is
+   * appended, with an inserted hyphen between these two parts. Therefore
+   * the given `name` argument does not need to be unique within the scope
+   * of the parent actor.
+   */
+  def spawnAdapter[U](f: U ⇒ T, name: String): ActorRef[U]
+
+  /**
+   * Create an anonymous child actor that will wrap messages such that other Actor’s
+   * protocols can be ingested by this Actor. You are strongly advised to cache
+   * these ActorRefs or to stop them when no longer needed.
+   */
+  def spawnAdapter[U](f: U ⇒ T): ActorRef[U] = spawnAdapter(f, "")
+
+}
+
+object Actor {
+  import Behavior._
+
+  // FIXME check that all behaviors can cope with not getting PreStart as first message
+
+  final implicit class BehaviorDecorators[T](val behavior: Behavior[T]) extends AnyVal {
+    /**
+     * Widen the wrapped Behavior by placing a funnel in front of it: the supplied
+     * PartialFunction decides which message to pull in (those that it is defined
+     * at) and may transform the incoming message to place them into the wrapped
+     * Behavior’s type hierarchy. Signals are not transformed.
+     *
+     * see also [[Actor.Widened]]
+     */
+    def widen[U](matcher: PartialFunction[U, T]): Behavior[U] = Widened(behavior, matcher)
+  }
+
+  private val _nullFun = (_: Any) ⇒ null
+  private def nullFun[T] = _nullFun.asInstanceOf[Any ⇒ T]
+  private implicit class ContextAs[T](val ctx: AC[T]) extends AnyVal {
+    def as[U] = ctx.asInstanceOf[AC[U]]
+  }
+
+  /**
+   * Widen the wrapped Behavior by placing a funnel in front of it: the supplied
+   * PartialFunction decides which message to pull in (those that it is defined
+   * at) and may transform the incoming message to place them into the wrapped
+   * Behavior’s type hierarchy. Signals are not transformed.
+   *
+   * Example:
+   * {{{
+   * Stateless[String]((ctx, msg) => println(msg)).widen[Number] {
+   *   case b: BigDecimal => s"BigDecimal($b)"
+   *   case i: BigInteger => s"BigInteger($i)"
+   *   // drop all other kinds of Number
+   * }
+   * }}}
+   */
+  final case class Widened[T, U](behavior: Behavior[T], matcher: PartialFunction[U, T]) extends Behavior[U] {
```

Comment:
yes, of course it could, but then it would not be possible for users to decompose behaviors by pattern matching—I don’t know how much weight this carries but it applies to the whole DSL right now